File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int len=nums.length;
        int[] ans;

        for(int i=0;i<len;i++){
            for(int j=i+1;j<len;j++){
                if(nums[i]+nums[j]==target){
                    ans=new int[]{i,j};
                    return ans;
                }
            }
        }
        ans=new int[2];
        return ans;

    }
}

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int length=nums.size();
        vector<int> ans;
        // stores values and indexes
        unordered_map<int,int> mpp;

        for(int i=0;i<length;i++){
            mpp[nums[i]]=i;
        }

        for(int i=0;i<length;i++){
            int reqd=target-nums[i];
            if(mpp.find(reqd)!=mpp.end() && mpp[reqd]!=i){
                return {i,mpp[reqd]};
            }
        }

        return {};
    }
};